src/index.js

```diff
@@ -45,13 +48,213 @@ module.exports = {
       }],
     }, {
       type: 'function',
-      fn(opts) { return localRunner.run(opts); },
+      fn(opts) { 
+      // if W doesn't merge, return run(), if W merge, return statistic()
+//        console.log('opts.remoteResult.data is',opts.remoteResult.data);
+//         console.log('opts.previousData is', opts.previousData);
+       console.log('opts.remoteResult.data.endOfIteration is' , opts.remoteResult.data.endOfIteration);
+       console.log('opts.remoteResult.data.statisticStep is', opts.remoteResult.data.statisticStep);
+        
+        if ((opts.remoteResult.data.endOfIteration == true) && (opts.remoteResult.data.statisticStep == 0)){
+          
+          // step 0 calculate local statistics and localYMean
+           const biasedX = opts.previousData.biasedX;
+           const y = opts.previousData.y;
+           const localCount = y.length;
+           const betaVector = regression.oneShot(biasedX, y);
+           const rSquared = regression.rSquared(biasedX, y, betaVector);
+           const tValue = regression.tValue(biasedX, y, betaVector);
+           var tdist = distributions.Studentt(localCount-1);
+           const tcdf = tValue.map(r => tdist.cdf(r));
+           const pValue = n.mul(2,n.sub(1,tcdf));
+           const localMeanY = n.sum(y) / localCount;
+ 
+           return {
+             betaVector,
+             localCount,
+             localMeanY,
+             rSquared,
+             tValue,
+             pValue,
+             biasedX,
+             y,
+             endOfIteration : true,
+             statisticStep : 0,
+          };          
+        }
+        else if ((opts.remoteResult.data.endOfIteration == true) && (opts.remoteResult.data.statisticStep == 1)){   
```

Comment:
The fact that we're branching on `statisticStep` makes me want to separate these into local computations, as mentioned before. We can at least move functionality into _src/local.js_ to clean up our computation file.